### ISSUE TITLE
(MIUI 14) Alarm does not wake up the screen

### ISSUE THREAD

**Issue by GaucovyAero** [vicolo-dev/chrono#106]
**Describe the bug**

**To Reproduce**
1. Enable all "Other permissions" for the app to be able to be shown on lock screen, turn off battery optimization and put the app to autostart as usual
2. Create an alarm in 1 minute, close the app and turn off the screen.
3. It does nothing, no display no sound, does not wake up the display
4. Wake up the display manually by clicking the power button and the alarm starts showing.

**Expected behavior**
Chrono should be able to wake up the display when an alarm comes up.

**Screenshots**
![Screenshot_2024-01-29-06-34-57-997_com miui securitycenter](https://github.com/vicolo-dev/chrono/assets/85581164/07970b67-2d2c-45ef-a573-3d3895219e0c)
![Screenshot_2024-01-29-06-35-06-236_com miui securitycenter](https://github.com/vicolo-dev/chrono/assets/85581164/73588f56-d500-4d80-9c14-e4355d0e49fc)

**Smartphone Information**
 - Device: Poco F3 8/256
 - OS: MIUI 14, Android 13
 - App version 0.2.10

**Additional context**
MIUI Optimizations are turned on, as turning them off creates severe visual bugs (white battery background instead of transparent, transparent backgrounds being pixelated, ...)

**Comment by GaucovyAero** [comment]
Behavior is a bit different on 0.3.0-rc2, but the app is still unusable for me
it shows the alarm for a split second and vibrates once if vibration is enabled
then the alarm closes as if it crashed and it sends me to the alarms menu (still on my device's lock screen), and thats all it does

**Comment by AhsanSarwar45** [comment]
@GaucovyAero Can you try with 0.3.0 release?

**Comment by GaucovyAero** [comment]
Okay, so, the crashing is fixed now, but there is another issue that might also be fixable. The alarm works by itself only when I leave the app running in the background. In order to make the alarm show up and sound when I don't have the app in the background, I have to wake up the screen by double tapping it. The simplemobiletools clock somehow has this figured out and works without issue even if I wipe the app from Recents.

**Comment by AhsanSarwar45** [comment]
Yes, I managed to reproduce the alarm not ringing until waking screen issue on a MiUI device, but not on any other. So seems like its a MiUI specific thing. Have you tried following all the instructions on [dontkillmyapp](https://dontkillmyapp.com/)? I will try doing the same on my device to see if that somehow fixes the issue. I'll also take a look at simplemobiletools clock to see how they solved it.

**Comment by GaucovyAero** [comment]
A lot of the instructions on there are for old MIUI, but what I dont remember trying is the autostart. I am currently not on MIUI, so if you can test it on your device, please do and report back :D

**Comment by AhsanSarwar45** [comment]
In my testing, this only happens rarely. Most of the times, the screen wakes just fine, whether app is in foreground, background, or closed, or whether the alarm rings 1 minute after closing screen, or 5 hours after. But this does this seem to happen from time to time. Tried with auto start and still reproduced this issue one time. Will keep looking.

**Comment by AhsanSarwar45** [comment]
Not sure exactly what I did, but this issues doesn't occur on my MIUI anymore. My guess is the Do not Disturb setting was doing something. Even thought DnD wasn't scheduled at that time, my research indicates it might still be causing issues. The other thing to look out for is to make sure the app has the 'show on lockscreen' permission. On my MIUI, this permission needs to be turned on in 2 separate places, one in the permissions section, and one in the notification section of app settings.

**Comment by GaucovyAero** [comment]
cool 👍, closing i guess :D

**Comment by WoodJasper** [comment]
this issue did help! with miui12 using now, i just found that i didn't permit chrono to show up from background and that's why chrono clock don't offer me buttons to dismiss it😂.
